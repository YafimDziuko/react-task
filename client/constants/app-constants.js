export const SEARCH = {
    genres: 'genres',
    title: 'title'
};

export const SORT = {
    rating: "vote_average",
    releaseDate: "release_date"
}

export const ACTIONS = {
    RETURN_TO_MAIN_PAGE: "RETURN_TO_MAIN_PAGE",
    UPDATE_INPUT_VALUE: "UPDATE_INPUT_VALUE",
    RESET_INPUT_VALUE: "RESET_INPUT_VALUE",
    SEARCH: "SEARCH",
    SWITCH_SEARCH: "SWITCH_SEARCH",
    SWITCH_SORT: "SWITCH_SORT",
    FULL_FILM_LOAD: "FULL_FILM_LOAD",
    PAGINATION_HANDLER: "PAGINATION_HANDLER"
}

export const ITEM_COUNT_PER_PAGE = 12;
export const PAGE_RANGE_DISPLAYED = 5;

export const INITIAL_STATE = {
    searchInfo: {},
    results: {
        data:[]
    },
    fullItem: {
        isActive: false
    },
    inputValue: "",
    sort: "releaseDate",
    search: 'title',
    resultsCount: -1
}