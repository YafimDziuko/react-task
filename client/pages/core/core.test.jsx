import React from 'react';
import configureStore from 'redux-mock-store';
import { fetch } from 'isomorphic-fetch';

import Core from './core';
import { INITIAL_STATE } from '../../constants/app-constants';

const mockStore = configureStore();
let store;

describe('<Core/>', function() {
  beforeEach(() => {
    store = mockStore(INITIAL_STATE);
  });

  it('should render Core and match snapshot', function() {
      const wrapper = shallow(<Core store={store}/>).dive();

      expect(wrapper).toMatchSnapshot();
  });

  it('should render Core with Full Film Item', function() {
    store = mockStore({...INITIAL_STATE, fullItem: { isActive: true }});
    const wrapper = shallow(<Core store={store}/>).dive();

    expect(wrapper).toMatchSnapshot();
  });
});

