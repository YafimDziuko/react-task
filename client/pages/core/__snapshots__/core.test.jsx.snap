// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Core/> should render Core and match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Core
    searchHandler={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
    user={
      Object {
        "fullItem": Object {
          "isActive": false,
        },
        "inputValue": "",
        "pagination": Object {
          "activePage": 1,
        },
        "results": Object {
          "data": Array [],
        },
        "resultsCount": -1,
        "search": "title",
        "searchInfo": Object {},
        "sort": "releaseDate",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(Header)
          fullItem={
            Object {
              "isActive": false,
            }
          }
        />,
        <Connect(ResultsBar) />,
        <Connect(ResultsBody)
          assets={
            Object {
              "data": Array [],
            }
          }
        />,
        <Footer />,
      ],
      "className": "core-page",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "fullItem": Object {
            "isActive": false,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "assets": Object {
            "data": Array [],
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(Header)
            fullItem={
              Object {
                "isActive": false,
              }
            }
          />,
          <Connect(ResultsBar) />,
          <Connect(ResultsBody)
            assets={
              Object {
                "data": Array [],
              }
            }
          />,
          <Footer />,
        ],
        "className": "core-page",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "fullItem": Object {
              "isActive": false,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "assets": Object {
              "data": Array [],
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<Core/> should render Core with Full Film Item 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Core
    searchHandler={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
    user={
      Object {
        "fullItem": Object {
          "isActive": true,
        },
        "inputValue": "",
        "pagination": Object {
          "activePage": 1,
        },
        "results": Object {
          "data": Array [],
        },
        "resultsCount": -1,
        "search": "title",
        "searchInfo": Object {},
        "sort": "releaseDate",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(Header)
          fullItem={
            Object {
              "isActive": true,
            }
          }
        />,
        <Connect(ResultsBar) />,
        <Connect(ResultsBody)
          assets={
            Object {
              "data": Array [],
            }
          }
        />,
        <Footer />,
      ],
      "className": "core-page",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "fullItem": Object {
            "isActive": true,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "assets": Object {
            "data": Array [],
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(Header)
            fullItem={
              Object {
                "isActive": true,
              }
            }
          />,
          <Connect(ResultsBar) />,
          <Connect(ResultsBody)
            assets={
              Object {
                "data": Array [],
              }
            }
          />,
          <Footer />,
        ],
        "className": "core-page",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "fullItem": Object {
              "isActive": true,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "assets": Object {
              "data": Array [],
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
